lib/rack/query_parser.rb

```diff
@@ -177,22 +177,42 @@ def key?(key)
         @params.key?(key)
       end
 
-      def to_params_hash
-        hash = @params
-        hash.keys.each do |key|
-          value = hash[key]
```

Comment:
The performance improvement is likely due to not having to allocate an array, together with the replacement of this hash index access.